service/src/main/java/org/ehrbase/dao/access/jooq/FolderAccess.java

```diff
@@ -511,17 +495,35 @@ public static I_FolderAccess getNewFolderAccessInstance(final I_DomainAccess dom
         return buildFolderAccessTreeRecursively(domainAccess, folder, null, dateTime, ehrId, null);
     }
 
+    /**
+     * {@inheritDoc}
+     * <br> Includes sub-folders.
+     */
     @Override
-    public Integer delete() {
-        // not valid, because missing mandatory data, like committer
-        throw new InternalServerException(CALLED_INVALID_ACCESS_LAYER_METHOD);
+    public int delete(LocalDateTime timestamp, UUID systemId, UUID committerId, String description) {
+        // create new contribution for this deletion action (with embedded contribution.audit handling)
+        contributionAccess = I_ContributionAccess.getInstance(getDataAccess(),
+            contributionAccess.getEhrId()); // overwrite old contribution with new one
+        var contribution = contributionAccess
+            .commit(TransactionTime.millis(), committerId, systemId, null,
+                ContributionDef.ContributionState.COMPLETE,
```

Comment:
@jakesmolka not sure if the ContributionState represents the VERSION.lifecycle_state of the VERSION inside the CONSITRIBUTION. If so, and this is a delete operation, the lifecycle_state should be DELETED. REF https://github.com/openEHR/terminology/blob/master/openEHR_RM/en/openehr_terminology.xml#L174

Reply:
See my previous comment for context.

Apart from that, also a good one. I'll document it in the issue, but won't touch the states right now due to the scope.